Nextcloud/Extensions/DependencyInjection.cs

```diff
@@ -17,77 +18,97 @@ public static IServiceCollection AddNextcloudClients(this IServiceCollection ser
         byte[] authToken = Encoding.UTF8.GetBytes($"{username}:{password}");
         var auth = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(authToken));
 
+
         services.AddHttpClient<INextcloudCircleClient, NextcloudCircleClient>(i =>
         {
             i.BaseAddress = new Uri(nextcloudUrl);
+            i.Timeout = TimeSpan.FromMinutes(5);
             i.DefaultRequestHeaders.Authorization = auth;
             i.DefaultRequestHeaders.UserAgent.ParseAdd(applicationName);
             i.DefaultRequestHeaders.Add("OCS-ApiRequest", "true");
             i.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-        });
+        }).AddResilienceHandler("rate-limit", GetRateLimiter);
 
-        CookieContainer circleCookieContainer = new CookieContainer();
         services.AddHttpClient<INextcloudCollectivesClient, NextcloudCollectivesClient>(i =>
         {
             i.BaseAddress = new Uri(nextcloudUrl);
+            i.Timeout = TimeSpan.FromMinutes(5);
             i.DefaultRequestHeaders.Authorization = auth;
             i.DefaultRequestHeaders.UserAgent.ParseAdd(applicationName);
             i.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            i.DefaultRequestHeaders.Add("requesttoken", NextcloudCollectivesClient.GetCsrfToken(i).Result);
         }).ConfigurePrimaryHttpMessageHandler(() => new HttpClientHandler()
         {
-            CookieContainer = circleCookieContainer,
             UseCookies = true,
-        });
+        }).AddResilienceHandler("rate-limit-1", i => i.AddConcurrencyLimiter(new ConcurrencyLimiterOptions()
+        {
+            PermitLimit = 1,
```

Comment:
Solle der Rate Limit nicht als Setting konfiguriert sein? Dann können wir per Umgebung die Rate anpassen (hoffentlich geht in prod mehr als 1)
```suggestion
            PermitLimit = 1,
```